clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -776,32 +778,23 @@ public PartitionRecords(long fetchOffset, TopicPartition partition, List<Consume
         }
 
         private boolean isEmpty() {
-            return records == null || records.isEmpty();
+            return records == null || position >= records.size();
```

Comment:
This function is probably poorly named. It's really used to indicate whether there are any records which haven't been returned to the user. Prior to this patch, the underlying collection always contained the unreturned records, but with this patch, it's the position that indicates which records are remaining.